DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Utils/ArcGisDocumentStore.cs

```diff
@@ -9,22 +14,81 @@ public class ArcGISDocumentStore : DocumentModelStore
   public ArcGISDocumentStore(JsonSerializerSettings serializerOption)
     : base(serializerOption)
   {
-    // POC: Subscribe here document related events like OnSave, OnClose, OnOpen etc...
     ActiveMapViewChangedEvent.Subscribe(OnMapViewChanged);
+    ProjectSavingEvent.Subscribe(OnProjectSaving);
+    ProjectClosingEvent.Subscribe(OnProjectClosing);
   }
 
+  private Task OnProjectClosing(ProjectClosingEventArgs arg)
+  {
+    if (MapView.Active is null)
+    {
+      return Task.CompletedTask;
+    }
+
+    WriteToFile();
```

Comment:
Just a question: do other connectors also save model cards onClose event? Meaning if user messed something up and didn't intent to save the file?

Reply:
We save this info even on doc switch. But this is smth we will reconsider later by just saving model cards right after they created instead on save or on close